core/field_image.js

```diff
@@ -50,6 +51,7 @@ Blockly.FieldImage = function(src, width, height, opt_alt) {
       this.height_ + 2 * Blockly.BlockSvg.INLINE_PADDING_Y);
   this.text_ = opt_alt || '';
   this.setValue(src);
+  this.clickHandler_ = onclick || null;
```

Comment:
Perhaps it would be a good idea to check here if the onclick argument passed is in fact a function, or a runtime error will occur if trying to execute a different type of variable.